game/songs.cc

```diff
@@ -117,34 +123,15 @@ void Songs::reload_internal() {
 	doneLoading = true;
 }
 
-const std::string SONGS_CACHE_JSON_FILE = "songs.json";
-
-void Songs::LoadCache() {
+Songs::Cache Songs::loadCache() {
 	const fs::path songsMetaFile = getCacheDir() / SONGS_CACHE_JSON_FILE;
 	auto jsonRoot = readJSON(songsMetaFile);
-	if (jsonRoot.empty()) return;
-	std::vector<std::string> allPaths;
-	for(const auto& songPaths : {getPathsConfig("paths/system-songs"), getPathsConfig("paths/songs")}) {
-		for(const auto& songPath: songPaths) {
-			allPaths.push_back(songPath.string());
-		}
-	}
-
-	for(auto const& song : jsonRoot) {
-		const auto songPath = song.at("txtFile").get<std::string>();
-		const bool isSongPathInConfiguredPaths = std::find_if(
-			allPaths.begin(),
-			allPaths.end(),
-			[songPath](const std::string& userSongItem) {
-				return songPath.find(userSongItem) != std::string::npos;
-			}) != allPaths.end();
-		STAT buffer;
-		if(_STAT(songPath.c_str(), &buffer) == 0 && isSongPathInConfiguredPaths) {
-			auto realSong = std::make_shared<Song>(song);
-			std::unique_lock<std::shared_mutex> l(m_mutex);
-			m_songs.push_back(std::move(realSong));
-		}
+	Cache cache;
+	for (auto const& songData : jsonRoot) {
+		auto song = std::make_shared<Song> (songData);
+		cache[song->filename] = std::move(song);
```

Comment:
Use `cache[song->filename.string()] = ...` to avoid compile errors on windows.

Reply:
OK.

Reply:
Nice,  works 👍